lib/iwdp.js

```diff
@@ -0,0 +1,81 @@
+import { exec, SubProcess } from 'teen_process';
+import EventEmitter from 'events';
+import logger from './logger';
+import request from 'request-promise';
+import { retryInterval } from 'asyncbox';
+
+const IWDP_CMD = 'ios_webkit_debug_proxy';
+const MAX_RETRIES = 10;
+
+class IWDP extends EventEmitter {
+
+  constructor (udid = null) {
+    super();
+    this.udid = udid;
+    this.attempts = 0;
+    this.port = 27753;
+    this.process = this.createIWDPProcess(); 
+    this.endpoint = `http://localhost:${this.port}`;
+  }
+
+  createIWDPProcess () {
+    let process = new SubProcess(IWDP_CMD, ['--config', `${this.udid}:${this.port},:${this.port + 1}-${this.port + 101}`]); // (see https://github.com/google/ios-webkit-debug-proxy for reference)// If the process exits and the exit wasn't requested by the API, restart it
+    process.on('exit', () => this.onExit());
+    return process;
+  }
+
+  async onExit () {
+    if (!this.exitRequested) {
+      this.process = this.createIWDPProcess();
+      await this.start(); 
+    }
+  }
+
+  async start () {
+    if (++this.attempts > MAX_RETRIES) {
```

Comment:
Case 2: When we intentionally stop IWDP a flag called `exitRequested` is set to `true` and then `process.stop()` is called which triggers `exit`. The callback for `exit` checks if `exitRequested === true` and if it does, it won't try restarting it.

Case 3: If the process is broken somehow and it it keeps exiting and we keep attempting to restart it, the tests will keep failing and the user won't know why. If we shut it down after 10 retries (10 was chosen arbitrarily, I could change that) then an error will be thrown and the user will be given logging information that explains the problem.